### DISCUSSION TITLE
Locked out from Two Factor Authentication.

### DISCUSSION THREAD

**Discussion by ghost** [community/community#22280] [New to GitHub]
<p>Hello <strong>, </strong> recently i had <strong>enable</strong> the Two Factor Authentication or 2Fa . I had done with the help of google authentication which i later camed to know its not at all reliable also i stored Recovery Codes which also keeps on changing when we change the security settings. </p>
<p>So right now both of them are not working, i am <strong>locked</strong> out from my account.</p>
<p>So to <strong>recover</strong> my account i used my  <strong>Facebook Token,</strong>  and also emailed the support for the same (to review my token as it was ready for review) </p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/1X/55a6f4d237a9a7063efdb8532142714e4a389bbd.png" data-download-href="/uploads/short-url/cdIhOj9tnCE6nqUI1t39SlNffLD.png?dl=1" title="Screenshot_1.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/10137/181070040-b69f0021-08e6-48a6-8c5f-a2876bcd7220.png" alt="Screenshot_1.png" data-base62-sha1="cdIhOj9tnCE6nqUI1t39SlNffLD" width="690" height="418" srcset="https://aws1.discourse-cdn.com/github/optimized/1X/55a6f4d237a9a7063efdb8532142714e4a389bbd_2_690x418.png, https://aws1.discourse-cdn.com/github/optimized/1X/55a6f4d237a9a7063efdb8532142714e4a389bbd_2_1035x627.png 1.5x, https://aws1.discourse-cdn.com/github/original/1X/55a6f4d237a9a7063efdb8532142714e4a389bbd.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/1X/55a6f4d237a9a7063efdb8532142714e4a389bbd_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screenshot_1.png</span><span class="informations">1057×641 33.2 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>its been <strong>2 days</strong> i have got no response.  so what should i consider?

**Comment by ghost** [comment, marked as answer]
<p>Finally got accessed to the account. This Post help me i think  Thanks community!</p>

**Comment by majormoses** [comment]
<p>&gt; Hello <strong>, </strong> recently i had  <strong>enable</strong>  the Two Factor Authentication or 2Fa . I had done with the help of google authentication which i later camed to know its not at all reliable</p>
<p>There may be some other issue going on with your phone related to time, I have been using and administering github as well other applications that use Google Authenticator for many years without issues. A lot of companies and individuals out there rely on Google Authenticator (multiple times) every day. Like anything nothing is perfect but I would encourage you to not jump to conclusions about the overall reliability of a product used by hundreds of thousands of people because of a very small number of issues. Quite a few other time based MFA applications out there are based off Google authenticator implementation.</p>
<p>&gt; also i stored Recovery Codes which also keeps on changing when we change the security settings.</p>
<p>What are you changing? From what I have observed the only time your codes will re-generate is when you login, go to <a href="https://github.com/settings/security" rel="nofollow noopener">https://github.com/settings/security</a>, then click “Show” under “Recovery codes” section, and then “Generate recovery codes”.</p>
<p>On a more general note I would reccommend using hardware based MFA solutions where possible. I have multiple <a href="https://www.yubico.com/" rel="nofollow noopener">yubikeys</a> which I can use one of them recovery should I lose or damage one of them. I use this as well as the software based app. I would also avoid SMS based (even as a fallback) solutions as they have been susceptible to issues for years. If you use facebook token for recovery you want to make sure you have properly secured your facebook account as that could be exploited by malicious actors to gain access to your github. While this may sound a little far fetched I have actually seen this happen with multiple companies to gain access in hopes of finding vulnerabilities in proprietary products, use it to launch an attack against applications/infrastructure if CD is present, etc.</p>
